--- drbd-headers/linux/genl_magic_func.h
+++ drbd-headers/linux/genl_magic_func.h
@@ -298,8 +298,12 @@ static int CONCAT_(GENL_MAGIC_FAMILY, _genl_multicast_ ## group)(	\
 {									\
 	unsigned int group_id =						\
 		CONCAT_(GENL_MAGIC_FAMILY, _group_ ## group);		\
-	return genlmsg_multicast_allns(&ZZZ_genl_family, skb, 0,	\
-				 group_id);				\
+	int ret;		 					\
+	rcu_read_lock();						\
+	ret = genlmsg_multicast_allns(&ZZZ_genl_family, skb, 0,		\
+			 group_id, GFP_ATOMIC);				\
+	rcu_read_unlock();						\
+	return ret;							\
 }

 #include GENL_MAGIC_INCLUDE_FILE
