--- drbd-headers/linux/genl_magic_func-genl_register_family_with_ops_groups.h
+++ drbd-headers/linux/genl_magic_func-genl_register_family_with_ops_groups.h
@@ -19,7 +19,7 @@ static int CONCAT_(GENL_MAGIC_FAMILY, _genl_multicast_ ## group)(	\
 	unsigned int group_id =						\
 		CONCAT_(GENL_MAGIC_FAMILY, _group_ ## group);		\
 	return genlmsg_multicast_allns(&ZZZ_genl_family, skb, 0,	\
-				 group_id);				\
+				 group_id, GFP_ATOMIC);			\
 }
 
 #include GENL_MAGIC_INCLUDE_FILE
