--- /usr/lib/rpm/redhat/kmodtool
+++ /usr/lib/rpm/redhat/kmodtool
@@ -65,12 +65,19 @@
 {
     local variant="${1}"
     local dashvariant="${variant:+-${variant}}"
+    local dotvariant="${variant:+.${variant}}"
+
     case "$verrel" in
         *.el*) kdep="kernel${dashvariant}-%{_target_cpu} = ${verrel}" ;;
         *.EL*) kdep="kernel${dashvariant}-%{_target_cpu} = ${verrel}" ;;
         *)     kdep="kernel-%{_target_cpu} = ${verrel}${variant}"     ;;
     esac
 
+    echo "%global verrel $verrel"
+    echo "%global variant ${variant:-%nil}"
+    echo "%global dashvariant ${dashvariant:-%nil}"
+    echo "%global dotvariant ${dotvariant:-%nil}"
+
     echo "%package       -n kmod-${kmod_name}${dashvariant}"
 
     if [ -z "$kmp_provides_summary" ]; then
@@ -100,7 +107,6 @@
     fi
     
     cat <<EOF
-Requires:         ${kmod_name}-kmod-common >= %{?epoch:%{epoch}:}%{version}
 Requires(post):   /sbin/depmod
 Requires(postun): /sbin/depmod
 EOF
